tests/Helpers/AuthHelper.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Uasoft\Badaso\Module\LMSModule\Tests\Helpers;
+
+use Exception;
+use Tests\TestCase;
+
+class AuthHelper
+{
+    protected static $loginUrl = '/admin/v1/auth/login';
+
+    protected static $cache = [];
+
+    public static function asUser(TestCase $testCase, $user)
+    {
+        if (!isset($user->rawPassword)) {
+            throw new Exception(
+                'Please set user raw password first, e.g. `$user->rawPassword = \'password\';`'
+            );
+        }
+
+        $cachedToken = self::getCachedUserToken($user);
+        if (!empty($cachedToken)) {
+            return $cachedToken;
+        }
+
+        $credentials = [
+            'email' => $user->email,
+            'password' => $user->rawPassword,
+        ];
+
+        $loginResponse = $testCase->json('POST', self::$loginUrl, $credentials);
+
+        $token = $loginResponse->json('accessToken');
+
+        self::setCachedUserToken($user, $token);
+
+        return $testCase->withHeader('Authorization', 'Bearer ' . $token);
+    }
+
+    public static function getCacheKey($user)
+    {
+        return $user->email . ' : ' . $user->id;
+    }
+
+    public static function getCachedUserToken($user)
+    {
+        $cacheKey = static::getCacheKey($user);
+        return self::$cache[$cacheKey] ?? NULL;
+    }
+
+    public static function setCachedUserToken($user, $token)
+    {
+        $cacheKey = static::getCacheKey($user);
+        self::$cache[$cacheKey] = $token;
+    }
+}
```

Comment:
does this auth helper have anything related to the login implementation, or are you only using this for the testing?

Reply:
yes, for testing only, since badaso didn't provide any helper to log an arbitrary user in our testcases. It only provides a helper to log in an admin user